src/components/Header/Header.js

```diff
@@ -0,0 +1,30 @@
+import { useContext } from 'react';
+import { NavLink } from 'react-router-dom';
+import { UserContext } from '../../context/UserContext';
+import './Header.css';
+
+
+function Header() {
+
+  const { user, setUser } = useContext(UserContext);
+
+  const handleLogout = async () => {
+    setUser(null);
+  };
+
+  return (
+    <>
+      <div className="header">
+        <NavLink to="/"><h2 className="nav title">Alchemy ToDos</h2></NavLink>
+        {user && (
+          <div>
+            <h2 className="nav greeting">Hello, {user.email}!</h2>
+            <NavLink to="/auth"><button onClick={handleLogout} className="nav button">Logout</button></NavLink>
```

Comment:
You'll want this NavLink to route to `/auth/sign-in` since no routes you have match just `/auth`